secboot/secboot_hooks.go

```diff
@@ -74,18 +74,10 @@ func SealKeysWithFDESetupHook(runHook fde.RunSetupHookFunc, keys []SealKeyReques
 }
 
 func writeKeyData(path string, keySetup *fde.InitialSetupResult, auxKey []byte, model sb.SnapModel) error {
-	var handle []byte
-	if keySetup.Handle == nil {
-		// this will reach fde-reveal-key as null but should be ok
-		handle = []byte("null")
```

Comment:
have we tested if this work or is not needed anymore? I suppose this is about trying things with the OptEE hooks we have

Reply:
The original commit: https://github.com/snapcore/snapd/pull/13210/commits/98076561744c024344e3a1c3266bd562ecb5362d

 secboot: secboot.sb.NewKeyData() now marhals the handle internally

The old version of secboot.sb.NewKeyData() was expecting a handle
in json encoded form. However the new code is now doing the
marshalling itself.

Reply:
Looking at the spike branch, something similar is also done there. It has to do with dereference. So for sure the code is tested with nil, since I had to deal with it.